operator_tracker/src/lib.rs

```
@@ -0,0 +1,175 @@
+    address: String,
+    pub version: String,
+}
+
+impl IntoResponse for OperatorVersion {
+    fn into_response(self) -> axum::http::Response<Body> {
+        axum::http::Response::builder()
+            .status(axum::http::StatusCode::OK)
+            .body(Body::from(serde_json::to_string(&self).unwrap()))
+            .unwrap()
+    }
+}
+
+pub async fn create_or_update_operator_version(
+    db: &PgPool,
+    registry_coordinator: &RegistryCoordinatorContract,
+    payload: OperatorVersionPayload,
+) -> Result<Option<OperatorVersion>, OperatorVersionError> {
+    debug!(
+        "Received operator version request. Version {}",
+        payload.version
+    );
+
+    // Recover operator address from signature
+
+    // hash keccak256(version) and recover address from signature
+    let version = keccak256(payload.version.as_bytes());
+
+    // check version matches v*.*.* format with regex
+    if !regex::Regex::new(r"^v\d+\.\d+\.\d+$")
+        .unwrap()
+        .is_match(&payload.version)
+    {
+        return Err(OperatorVersionError::InvalidSignature);
+    }
+
+    // decode base64 signature
+    let signature = BASE64_STANDARD
+        .decode(payload.signature)
+        .map_err(|_| OperatorVersionError::InvalidSignature)?;
+
+    if signature.len() != 65 {
+        return Err(OperatorVersionError::InvalidSignature);
+    }
+
+    let signature = Signature {
+        r: ethers::types::U256::from_big_endian(&signature[0..32]),
+        s: ethers::types::U256::from_big_endian(&signature[32..64]),
+        v: signature[64] as u64,
+    };
+
+    let operator_address = signature
+        .recover(version)
+        .map_err(|_| OperatorVersionError::InvalidSignature)?;
+
+    info!("Operator address: {:?}", operator_address);
+
+    // Check if operator is registered on registry coordinator
+    let status = registry_coordinator
+        .get_operator_status(operator_address)
+        .await
+        .expect("Failed to get operator status");
```

let's return an error here